### PR TITLE
Update amplify model, add loss tests

### PR DESCRIPTION
**Pull request by pstjohn**
Not sure we support changing configs like this, but this would fail without handling the non-swiglu variant

## Summary by CodeRabbit

- **Bug Fixes**
  - Fixes feed‑forward sizing for non‑SwiGLU activations and prevents initialization/runtime errors, improving stability across activation options.
  - Note: forward method signatures were simplified (removed variable kwargs), which changes the public forward API.

- **Tests**
  - Adds deterministic masking seed and four loss/regression tests validating model behavior for pretrained and reinitialized variants.

- **Chores**
  - Updates development container setup and install/run configuration for the project.

### CODE REVIEW

File: models/amplify/tests/test_amplify_model.py [hunk 1]
```diff
@@ -168,3 +168,52 @@ def test_convert_state_dict():
     te_state_dict_keys.remove("decoder.bias")
 
     assert len(te_state_dict_keys) == 0
+
+
```
Comment (jomitchellnv) [inline models/amplify/tests/test_amplify_model.py:172]: needs gpu pytest mark parametrize

File: models/amplify/tests/test_amplify_model.py [hunk 2]
```diff
@@ -168,3 +168,52 @@ def test_convert_state_dict():
...
+    model.to("cuda", dtype=torch.bfloat16)
+    input_data = {k: v.to("cuda") for k, v in input_data.items()}
+    model.eval()
+    with torch.amp.autocast(device_type="cuda", dtype=torch.bfloat16):
+        output = model(**input_data)
+
+    torch.testing.assert_close(output.loss.detach().cpu(), torch.tensor(2.4), atol=1e-1, rtol=1e-2)
+
+
+def test_te_trained_model_loss(input_data):
```
Comment (jomitchellnv) [inline models/amplify/tests/test_amplify_model.py:184]: needs gpu

**Review by jomitchellnv — APPROVED** [review]
Coderabbit is awesome. What was the error with KWargs and Amplify?

File: models/amplify/tests/test_amplify_model.py (hunk 2 above)
Comment (pstjohn) [reply models/amplify/tests/test_amplify_model.py:184]: Everything needs a GPU currently, let's go back and annotate if we end up having a split CI where we can test some of this on CPU only workers

**Comment by pstjohn** [comment]
> Coderabbit is awesome. What was the error with KWargs and Amplify? 

If you had kwargs in your models forward pass, accelerate will scale your loss by the number of processes per node.
